src/lib.rs

```diff
@@ -10,7 +10,7 @@ use utils::*;
 
 #[derive(Debug, Clone, PartialEq)]
 pub enum Node {
-    Hash(Vec<u8>, usize), // (Hash, # empty spaces)
+    Hash { hash: Vec<u8>, spaces: usize },
```

Comment:
@gballet I think this improves readability. If you agree, I'll change the rest, if not I'll close this.

Reply:
If it's not too tedious, please do!